tests/integration/google_cloud_storage/test_google_cloud_storage.py

```diff
@@ -102,46 +106,99 @@ def _setup_profile(self, profile_auth_by):
             if use_datastore_profile
             else "gcs://" + self._bucket_name
         )
-        self._object_url = self._bucket_path + "/" + self._object_path
+        self.run_dir_url = f"{self._bucket_path}/{self.run_dir}"
+        self._object_url = self.run_dir_url + "/" + object_file
         logger.info(f"Object URL: {self._object_url}")
+        self.storage_options = {}
 
-    def teardown_class(self):
-        self.clean_test_directory(
-            bucket_name=self._bucket_name,
-            object_dir=self.object_dir,
-            gcs_fs=self._gcs_fs,
-        )
+    @classmethod
+    def teardown_class(cls):
+        cls.clean_test_directory()
 
-    def _perform_google_cloud_storage_tests(self, secrets={}):
-        data_item = mlrun.run.get_dataitem(self._object_url, secrets=secrets)
-        data_item.put(test_string)
+    def _setup_by_google_credentials_file(self, use_datastore_profile, use_secrets):
+        # We give priority to profiles, then to secrets, and finally to environment variables.
+        self.storage_options = {}
+        if use_datastore_profile:
+            self._setup_profile(profile_auth_by="credentials_json_file")
+            if use_secrets:
+                self.storage_options = {"GOOGLE_APPLICATION_CREDENTIALS": "wrong path"}
+            else:
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "wrong path"
+        else:
+            if use_secrets:
+                self.storage_options = {
+                    "GOOGLE_APPLICATION_CREDENTIALS": self.credentials_path
+                }
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = "wrong path"
+            else:
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = self.credentials_path
+
+    def _setup_by_serialized_json_content(self, use_datastore_profile, use_secrets):
+        self.storage_options = {}
+        if use_datastore_profile:
+            self._setup_profile(profile_auth_by="gcp_credentials")
+            if use_secrets:
+                self.storage_options = {"GCP_CREDENTIALS": "wrong credentials"}
+            else:
+                os.environ["GCP_CREDENTIALS"] = "wrong credentials"
+        else:
+            if use_secrets:
+                self.storage_options = {"GCP_CREDENTIALS": self.credentials}
+                os.environ["GCP_CREDENTIALS"] = "wrong credentials"
+            else:
+                os.environ["GCP_CREDENTIALS"] = self.credentials
+
+    @pytest.mark.parametrize(
+        "setup_by, use_secrets",
+        [
+            ("credentials_file", False),
+            ("credentials_file", True),
+            ("serialized_json", False),
+            ("serialized_json", True),
+        ],
+    )
+    def test_perform_google_cloud_storage_tests(
+        self, use_datastore_profile, setup_by, use_secrets
+    ):
+        #  TODO split to smaller tests by datastore conventions.
+        self.setup_mapping[setup_by](self, use_datastore_profile, use_secrets)
```

Comment:
This mechanism doesn't really make sense to me... It's a global setup, that registers a datastore profile and sets environment variables, but these are never deregistered or unset. So the leftovers from a run with one parameter set can affect subsequent runs.

Reply:
os.environ reset every run in `setup_before_each_test`.
Regarding profiles - because they have the same profile name, it gets overwritten every time it runs (every time we create a new profile), but I take your comment and I added a remove to the profiles by profile_name.